lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -0,0 +1,353 @@
+        dry air and water vapour in proportion given by the specific humidity.
+
+        Args:
+            mixing_ratio : iris.cube.Cube
+                Cube of specific humidity (fractional).
+        Returns:
+            iris.cube.Cube
+                Specific heat capacity of moist air (J kg-1 K-1).
+        """
+        specific_heat = ((-1.*mixing_ratio + 1.) * cc.U_CP_DRY_AIR +
+                         mixing_ratio * cc.U_CP_WATER_VAPOUR)
+        specific_heat.rename('specific_heat')
+        return specific_heat
+
+    @staticmethod
+    def latent_heat_of_condensation(temperature_input):
+        """
+        Calculate a temperature adjusted latent heat of condensation for water
+        vapour using the relationship employed by the UM.
+
+        Args:
+            temperature : iris.cube.Cube
+                A cube of air temperatures (Celsius, converted if not).
+        Returns:
+            iris.cube.Cube
+                Temperature adjusted latent heat of condesation (J kg-1).
+        """
+        temperature = temperature_input.copy()
+        temperature.convert_units('celsius')
+        latent_heat = (-1. * cc.U_LATENT_HEAT_T_DEPENDENCE * temperature +
+                       cc.U_LH_CONDENSATION_WATER)
+        latent_heat.units = cc.U_LH_CONDENSATION_WATER.units
+        latent_heat.rename('latent_heat_of_condensation')
+        return latent_heat
+
+    @staticmethod
+    def calculate_enthalpy(mixing_ratio, specific_heat, latent_heat,
+                           temperature):
+        """
+        Calculate the enthalpy (total energy per unit mass) of air (J kg-1).
+
+        Args:
+            mixing_ratio : iris.cube.Cube
+                Cube of mixing ratios.
+            specific_heat : iris.cube.Cube
+                Cube of specific heat capacities of moist air (J kg-1 K-1).
+            latent_heat : iris.cube.Cube
+                Cube of latent heats of condensation of water vapour
+                (J kg-1).
+            temperature : iris.cube.Cube
+                A cube of air temperatures (K).
+        Returns:
+           enthalpy : iris.cube.Cube
+               A cube of enthalpy values calculated at the same points as the
+               input cubes (J kg-1).
+        """
+        enthalpy = latent_heat * mixing_ratio + specific_heat * temperature
+        enthalpy.rename('enthalpy_of_air')
+        return enthalpy
+
+    @staticmethod
+    def calculate_d_enthalpy_dt(mixing_ratio, specific_heat,
+                                latent_heat, wet_bulb_temperature):
+        """
+        Calculate the enthalpy gradient with respect to temperature.
+
+        Args:
+
+        Returns:
+
+        """
+        wet_bulb_temperature.convert_units('K')
+        numerator = (mixing_ratio * latent_heat ** 2)
+    #    R_term = np.max(np.finfo(cc.R_WATER_VAPOUR).eps,
+    #                    cc.R_WATER_VAPOUR * wet_bulb_temperature ** 2)
```

Comment:
Why is this comment here?

Reply:
In the UM they have such a line to deal with numbers falling to or below zero.  I put it here as a place holder to think about it. I don't see a situation in which wet bulb temperatures can get close to zero, so I have removed it.